Fix typo in interface list for v1.0

cniethammer opened the pull request:
The duplicate entry for PMIx_Lookup should be PMIx_Lookup_nb, which existed in v1.0 according to the version mark.

jjhursey commented:
@cniethammer Thanks for the patch! It looks like you did not sign off your commit. Can you re-push with a `Signed-off-by: ...` (with your name and email address in place of the `...`) line in the commit message? If you need help just let us know.

cniethammer commented:
@jjhursey Thanks, I was not aware, that for the standard I have add a signed-off by.
I added the "signed-off by" and also rebased on the latest master version of the document.

jjhursey commented:
No problem. We should make that more clear in the PR template (I'll make a note to fix that).
Thanks for the fix!
